core/trie.go

```diff
@@ -15,6 +15,10 @@ const (
 
 	// AutoBalanceEnabled is used for data tries, and only after the activation of AutoBalanceDataTriesEnableEpoch flag
 	AutoBalanceEnabled
+
+	// WithoutCodeLeaf is used for account with code, it specifies that the trie code leaf has been moved to storage,
+	// it is enabled only after the activation of MigrateCodeLeafEnableEpoch flag
+	WithoutCodeLeaf
```

Comment:
all nodes with version "without code leaf" are assumed to be also with auto balance enabled right?

Just to confirm, it would not be possible to move from NotSpecified directly to WithoutCodeLeaf right?
If this is the case we should take care of the testing process to allow enough time (and actively do the first migration - auto balance) before this and ensure everything is migrated to auto balance before, or directly start with AutoBalanceEnabled active.

Also would this version change affect the logic for auto balancing since the version will be different but auto balancing will still need to be done? (I am asking here if there are checks in the code for the specific AutoBalanceEnabled version)